### PR TITLE
Rename Occlum attester to SGX attester and add Gramine support to it

### PR DESCRIPTION
**Pull request by mythi**
WIP: I'm planning to change the existing `occlum-attester` to `sgx-dcap-attester`.

TODO:

- [x] rename feature names
- [x] rename structs/modules
- [x] rebase this PR

### CODE REVIEW

**Comment by mythi** [comment]
/cc @Xynnn007 

I've tested both Gramine and Occlum paths manually. Occlum tester is also fixed and can be added back to CI.

File: attester/src/sgx_dcap/mod.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        SgxLibOsType::Invalid => false,
+        SgxLibOsType::Occlum => true,
+        SgxLibOsType::Gramine => true,
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+struct SgxDcapAttesterEvidence {
+    /// Base64 encoded SGX quote.
+    quote: String,
```
Comment (Xynnn007) [inline attester/src/sgx_dcap/mod.rs:41]: Do we need to add another field here named `libos_type` to indicate the type of libos? Though the integrity of this field is not protected.

**Review by Xynnn007 — APPROVED** [review]
LGTM and I like the reorganization.

File: attester/src/sgx_dcap/mod.rs (hunk 1 above)
Comment (mythi) [reply attester/src/sgx_dcap/mod.rs:41]: In my first version I had it for `SgxDcapAttester` and then `Defaulter` implemented for it that initialized it but then dropped that version. The verifier can use `quote` regardless of who generated it.

Comment (Xynnn007) [reply attester/src/sgx_dcap/mod.rs:41]: Well. I'm thinking about how the verifier choose the golden value of the MRENCLAVE.
1. With extra `libos_type`: There should be two lists of golden values for occlum and gramine agent-enclave, and they can be queried due to different `libos_type`
2. We only keep one list of both golden values for occlum and gramine agent-enclave. When verifying the quote, attestation service will use the values in the list to try to match the one in quote.

Both are fine with me.

Comment (mythi) [reply attester/src/sgx_dcap/mod.rs:41]: OK I see what you mean. I'd keep this PR as it is now and think about the quote verification for Gramine/Occlum co-existence later.

**Comment by mythi** [comment]
moved to `image-rs`

**Pull request closed without merging**